AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright 2018-2019 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSCore
+
+public class AmplifyAWSServiceConfiguration: AWSServiceConfiguration {
+    static let baseUserAgent: String = {
+        let version = "0.9.0"
+        let sdkName = "amplify-iOS"
+        let systemName = UIDevice.current.systemName.replacingOccurrences(of: " ", with: "-")
+        let systemVersion = UIDevice.current.systemVersion
+        let localeIdentifier = Locale.current.identifier
+        return "\(sdkName)/\(version) \(systemName)/\(systemVersion) \(localeIdentifier)"
+    }()
+
+    override public var userAgent: String {
+        return AmplifyAWSServiceConfiguration.baseUserAgent
+    }
+
+    override public func copy(with zone: NSZone? = nil) -> Any {
+        return super.copy(with: zone)
+    }
+
+    override init() {
+        super.init(region: .Unknown, credentialsProvider: nil)
+    }
+
+    override public init(region regionType: AWSRegionType,
+                         credentialsProvider: AWSCredentialsProvider!) {
```

Comment:
why with a `!` here on `AWSCredentialsProvider`?

Reply:
to be honest, this was copy and pasted it in from Jithin.. but looking into it, it looks like it's calling super.init() which is calling:
```
- (instancetype)initWithRegion:(AWSRegionType)regionType
           credentialsProvider:(id<AWSCredentialsProvider>)credentialsProvider;
```

That being said, i think you're right in that we can get away with removing it.